src/main/java/org/xstefank/model/PropertiesLoader.java

```diff
@@ -0,0 +1,67 @@
+package org.xstefank.model;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.xstefank.model.Utils.JBOSS_CONFIG_DIR;
+import static org.xstefank.model.Utils.CONFIG_FILE;
+
+public class PropertiesLoader {
+
+    private static PropertiesLoader propertiesLoader = null;
+    private Properties properties;
+
+    private PropertiesLoader() {
+        String dirProperty = System.getProperty(JBOSS_CONFIG_DIR);
+        if (dirProperty != null) {
+            File dir = new File(System.getProperty(JBOSS_CONFIG_DIR));
```

Comment:
why do you call `System.getProperty` again? you have the result stored in `dirProperty`